### DISCUSSION TITLE
Tripped TPM

### DISCUSSION THREAD

**Discussion by JerboaGobi** [noahbliss/mortar#27] [General]
Just wondering what the steps are to re-enroll TPM if it get's tripped.

**Comment by noahbliss** [comment]
This is a good question and I'll leave it open until the Readme gets updated. In short you'll do the following:

If you know the tpm password/can still write to it:  
Just rerun step 3.  

If you changed tpm modules/need to start over with binding:  
Go into the bios and reset/clear the tpm. Ensure it is in the setup mode afterwards.  
Reboot, enter disk password, rerun step 3.  


It should be that simple.

**Comment by JerboaGobi** [comment]
Thanks for confirming that. I actually intentionally tripped TPM and verified this myself. The process is essentially the same if UEFI verification gets tripped as well. I'm using a software raid one, both for the efi and lvm volumes and again intentionally wiped a drive and rebuilt the array doing so caused UEFI verification to fail and just rewriting the keys to secure boot and TPM sets everything back as expected. Just wanted to make sure I wasn't going to lock myself out when something happens. I actually wrote up documentation for the mdadm raid 1 array(s) I have set up, running Proxmox if you'd like to add that to the projects' documentation I'm happy to share it with you.

**Reply by noahbliss** [reply]
If you have it written up, feel free to send it here or via other means. Happy to take a look!
